wandb/sdk/wandb_run.py

```diff
@@ -2957,6 +2957,37 @@ def finish_artifact(
             finalize=True,
         )
 
+    def _download_artifact(
+        self,
+        artifact: Artifact,
+        root: Optional[str] = None,
+        recursive: bool = False,
+        allow_missing_references: bool = False,
+    ) -> FilePathStr:
+        if self._backend and self._backend.interface:
+            if not self._settings._offline:
+                future = self._backend.interface.download_artifact(
+                    artifact.qualified_name,
+                    root,
+                    recursive,
+                    allow_missing_references,
+                )
+                if hasattr(future, "response"):
+                    if future.response.download_artifact_response.error_message:
+                        raise ValueError(
+                            f"Error downloading artifact: {future.response.download_artifact_response.error_message}"
+                        )
+                    download_path = (
+                        future.response.download_artifact_response.file_download_path
+                    )
+                    return FilePathStr(download_path)
+                return FilePathStr("")
```

Comment:
I would assume `response` is always set on the `Result` and get rid of the if on line 2975.